hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -2145,37 +2168,51 @@ public OmKeyLocationInfo allocateBlock(OmKeyArgs args, long clientID,
    */
   @Override
   public OmKeyInfo lookupKey(OmKeyArgs args) throws IOException {
+    ResolvedBucket bucket = resolveBucketLink(args);
```

Comment:
Now for each operation if it is link bucket where if it has sourceVolume/SourceBucket we do checkAcls twice. One with READ permission on sourceBucket/SourceVolume. It is not clear why do we need that. 

If it requires, do you think we need to have an API where we can check all required ACLS with a single checkAcl call?

Reply:
Okay, I see is in resolveBucketLink we check Acls for read on provided bucket, and in actual request check acls on sourceBucket/sourceVolume which it is resolved to. Why do we need this?

Reply:
Read permission on the link is required to follow it.  Do you propose to completely skip ACL on link and allow anyone to use it?

Reply:
I am not sure what is the expected behavior, how is  this semantics derived?
Because with this approach for all mounted buckets, we do 2 checkAcls, we might put pressure on Ranger.

Can we rely on underlying bucket acls, as anyway we verify already. Any downside/security issue?

Reply:
Any info on this question?

Reply:
> I am not sure what is the expected behavior, how is this semantics derived?
> Can we rely on underlying bucket acls, as anyway we verify already. Any downside/security issue?

This matches unix symlinks permissions: if you have read access on the link, you see where it points to and can follow it.  ACL was discussed in design doc update (#1009) and the current behavior was approved.  Please check with @arp7 if we need any changes.  Also pinging @ChenSammi for input.

Reply:
> This matches unix symlinks permissions: if you have read access on the link, you see where it points to and can follow it. 
+1